### PR TITLE
Revert "Fix ordering of the promotion [Take II]"

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Reverts SciML/DiffEqBase.jl#1032. Sees if that fixes some of the Tracker tests that started failing.

### CODE REVIEW

File: ext/DiffEqBaseTrackerExt.jl [hunk 1]
```diff
@@ -97,15 +97,9 @@ Tracker.@grad function DiffEqBase.solve_up(prob,
...
-    sol, pb_f = DiffEqBase._solve_adjoint(
-        prob, sensealg, Tracker.data(u0), Tracker.data(p),
+    out = DiffEqBase._solve_adjoint(prob, sensealg, Tracker.data(u0), Tracker.data(p),
         SciMLBase.TrackerOriginator(), args...; kwargs...)
-
-    if sol isa AbstractArray
-        !hasfield(typeof(sol), :u) && return sol, pb_f # being safe here
-        return sol.u, pb_f # AbstractNoTimeSolution isa AbstractArray
-    end
-    return convert(AbstractArray, sol), pb_f
```
Comment (avik-pal) [inline ext/DiffEqBaseTrackerExt.jl:108]: probably reverting just this will do

**Comment by ChrisRackauckas** [comment]
Can you make that PR? I'm out and about right now.

File: ext/DiffEqBaseTrackerExt.jl (hunk 1 above)
Comment (avik-pal) [reply ext/DiffEqBaseTrackerExt.jl:108]: Let's wait and see https://github.com/SciML/DiffEqBase.jl/pull/1076

**Comment by avik-pal** [comment]
Source of the errors doesn't seem to be that PR. tests are still failing https://github.com/SciML/DiffEqBase.jl/actions/runs/10584877403/job/29330163818?pr=1075#step:6:1828

**Comment by avik-pal** [comment]
@ChrisRackauckas based on the stacktrace I have a feeling this is an OrdinaryDiffEq issue. I would expect the integrator cache to hold TrackedArrays and not regular arrays when using discrete adjoints.

**Comment by ChrisRackauckas** [comment]
I was starting to fear that it might be related to the OrdinaryDiffEq split PRs. But I don't see how? 

https://github.com/search?q=repo%3ASciML%2FOrdinaryDiffEq.jl%20value&type=code

Everything is directly using DiffEqBase.value, which is the thing that should be de-tracking as appropriate.

**Pull request closed without merging**